lib/common/filters/retrypolicyfilter.js

```diff
@@ -130,7 +130,7 @@ RetryPolicyFilter._handle = function (self, requestOptions, next) {
         // we should NOT retry within the SDK as the stream data is not valid anymore if we retry directly.
         if (!returnObject.outputStreamSent && returnObject.error && azureutil.objectIsNull(returnObject.retryable) &&
              ((!azureutil.objectIsNull(returnObject.response) && retryInfo.retryable) || 
-               (returnObject.error.code === 'ETIMEDOUT' || returnObject.error.code === 'ESOCKETTIMEDOUT' || returnObject.error.code === 'ECONNRESET' || returnObject.error.code === 'EAI_AGAIN'))) {
+               (returnObject.error.code === 'ECONNREFUSED' || returnObject.error.code === 'ENOTFOUND' || returnObject.error.code === 'ETIMEDOUT' || returnObject.error.code === 'ESOCKETTIMEDOUT' || returnObject.error.code === 'ECONNRESET' || returnObject.error.code === 'EAI_AGAIN'))) {
```

Comment:
For ENOTFOUND, we should only retry if it happens for the secondary endpoint, for primary endpoint, we should NOT retry it.